src/java.base/share/classes/javax/crypto/KDF.java

```diff
@@ -0,0 +1,631 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package javax.crypto;
+
+import jdk.internal.javac.PreviewFeature;
+import sun.security.jca.GetInstance;
+import sun.security.jca.GetInstance.Instance;
+import sun.security.util.Debug;
+
+import java.security.InvalidAlgorithmParameterException;
+import java.security.KDFParameters;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.Provider;
+import java.security.Provider.Service;
+import java.security.ProviderException;
+import java.security.spec.AlgorithmParameterSpec;
+import java.util.Iterator;
+import java.util.Objects;
+
+/**
+ * This class provides the functionality of a Key Derivation Function (KDF),
+ * which is a cryptographic algorithm for deriving additional keys from input
+ * keying material and (optionally) other data.
+ * <p>
+ * {@code KDF} objects are instantiated with the {@code getInstance} family
+ * of methods. For <a href="https://www.rfc-editor.org/rfc/rfc5869">RFC 5869</a>,
+ * KDF algorithm names follow a naming convention of
+ * <em>Algorithm</em>With<em>PRF</em>. For instance, a KDF implementation of
```

Comment:
I only see HKDF mentioned in RFC 5869. There is no naming convention of <Algorithm>With<PRF> anywhere. Is there any other spec/standard or crypto libraries using this naming?

Reply:
Yes, I am aware of the Cipher transformations, etc. It's just that the sentence seems to imply the naming convention is from RFC 5869.

Reply:
Yes.
However I still question the current naming convention of using HmacXXX with HKDF.

Reply:
I think this issue needs more discussion. It is really important to reach consensus on the algorithm name format. I don't have an opinion yet, but will think about it.